tensorflow/core/kernels/unravel_index_op.cc

```diff
@@ -54,6 +54,24 @@ class UnravelIndexOp : public OpKernel {
 
     auto dims = dims_tensor.vec<Tidx>();
 
+    // Chek to make sure indices is not out of boundary
+    Eigen::Tensor<bool, 0, Eigen::RowMajor> check;
+    if (TensorShapeUtils::IsScalar(indices_tensor.shape())) {
+      auto indices = indices_tensor.scalar<Tidx>();
+      auto dims_prod = dims.prod();
+      check = (indices < dims_prod).all();
+    } else {
+      auto indices = indices_tensor.vec<Tidx>();
+      auto dims_prod = dims.prod()
+          .reshape(Eigen::array<Eigen::Index, 1>({1}))
+          .broadcast(
+              Eigen::array<Eigen::Index, 1>({indices_tensor.NumElements()}));
+      check = (indices < dims_prod).all();
+    }
```

Comment:
Looks like Eigen is abused here :) This can be written with a `std::for_all` over "pointer-range".

```
Tidx* indices = indices_tensor.data() // is it available?
or
Tidx* indices = indices_tensor.scalar<Tidx>().data() // or this, don't remember the details.

bool check = std::for_all(indices, indicex + size, [&](Tidx index) { return index < dims_prod; })
```